Kohei Noda

kohei-noda-qcrg opened the pull request.

kohei-noda-qcrg committed:
Add ipython to requirements.txt

kohei-noda-qcrg committed:
Line length limit is 500.

kohei-noda-qcrg committed:
Implemented _get_user_by_token and update_user.

kohei-noda-qcrg committed:
Add /user/update test

kohei-noda-qcrg committed:
WIP: create_room

kohei-noda-qcrg committed:
Merge pull request #1 from kohei-noda-qcrg/dev

WIP: bigint

methane commented on schema.sql:
```diff
@@ -9,3 +9,20 @@ CREATE TABLE `user` (
+  `leader_id` bigint NOT NULL,
+  `joined_user_count` int NOT NULL,
+  PRIMARY KEY (`room_id`)
+)
```
```suggestion
);
```

kohei-noda-qcrg replied:
レビューいただきありがとうございます。

mysql < schema.sql
でスキーマを流しこんだときに
drop table room_userのところでエラーが出て、不思議に思っていたのですが、
セミコロン忘れだったんですね。
気がつきませんでした...

kohei-noda-qcrg committed:
Update schema.sql

Co-authored-by: Inada Naoki <songofacandy@gmail.com>

kohei-noda-qcrg committed:
Add a max_user_count column to the room table.

kohei-noda-qcrg committed:
Add the /room/list API.

kohei-noda-qcrg committed:
Comment out unimplemented tests

kohei-noda-qcrg committed:
Merge pull request #2 from kohei-noda-qcrg:room_list

Add the /room/list API

kohei-noda-qcrg committed:
Add the /room/join API.

kohei-noda-qcrg committed:
Merge pull request #3 from kohei-noda-qcrg:room_join

Add the /room/join API

kohei-noda-qcrg committed:
The return type of join_room is JoinRoomResult.

kohei-noda-qcrg committed:
Return the join_room_result

kohei-noda-qcrg committed:
Add the /room/wait API

kohei-noda-qcrg committed:
Explicitly import classes from model.py

kohei-noda-qcrg committed:
Merge pull request #4 from kohei-noda-qcrg:room_wait

Add the /room/wait API

kohei-noda-qcrg committed:
Add the /room/start API

kohei-noda-qcrg committed:
Merge pull request #5 from kohei-noda-qcrg:room_start

Add the /room/start API

kohei-noda-qcrg committed:
Fix room_start UPDATE

kohei-noda-qcrg committed:
Add the score column

kohei-noda-qcrg committed:
Add the /room/end API

kohei-noda-qcrg committed:
Merge pull request #6 from kohei-noda-qcrg:room_end

Add the /room/end API

kohei-noda-qcrg committed:
Fix the descriptions about APIs

kohei-noda-qcrg committed:
Use json for judge_count_dict

kohei-noda-qcrg committed:
Add the /room/result API.

kohei-noda-qcrg committed:
Merge pull request #7 from kohei-noda-qcrg/room_result

Add the /room/result API.

kohei-noda-qcrg committed:
Merge remote-tracking branch 'upstream/main'

kohei-noda-qcrg committed:
Add /room/leave

kohei-noda-qcrg committed:
Merge pull request #8 from kohei-noda-qcrg:room_leave

Add the /room/leave API

methane commented on app/model.py:
```diff
@@ -5,6 +5,7 @@
 from pydantic import BaseModel
 from sqlalchemy import text
 from sqlalchemy.exc import NoResultFound
+from sqlalchemy.future import Connection
```
futureはSQLAlchemy 1.4の頃に2.0の機能を使うときに使っていたパッケージの名前なので、今は次のように書いた方がいいと思います。

```py
from sqlalchemy.engine import Connection
```

kohei-noda-qcrg replied:
ありがとうございます！
修正しました

kohei-noda-qcrg committed:
refactor

kohei-noda-qcrg committed:
Use _update_wait_room_status (room_start)

kohei-noda-qcrg committed:
Update wait_room_status to Dissolution (/room/result).

If host calls /room/result and
result is not empty list,
update wait_room_status to Dissolution.

kohei-noda-qcrg committed:
Add timeout (3min)
